### PR TITLE
DOC: Clarify behavior for Series with dict-like data and index

### LINKED ISSUE #34534: BUG: pd.Series produces NaN for valid input as per documentation
- [x] I have checked that this issue has not already been reported.

- [ ] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample

```python
import pandas as pd

dept = {'CS': 'Engineering',
        'ENG': 'Arts',
        'PHY': 'Science'}

dept = pd.Series(dept, index=['A', 'B', 'C'])
print(dept)
```

#### Problem description

pandas series is created by passing dict as input data and also index as additional argument. The indexes are updated but the values are changed to NaN. Whereas if I remove the index argument, it works fine.

#### Expected Output
CS     Engineering
ENG           Arts
PHY        Science
dtype: object
#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.5.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
machine          : AMD64
processor        : Intel64 Family 6 Model 69 Stepping 1, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : English_India.1252

pandas           : 0.25.3
numpy            : 1.18.1
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 19.3.1
setuptools       : 41.2.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.4.2
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.10.3
IPython          : 7.11.1
pandas_datareader: None
bs4              : 4.8.2
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : 4.4.2
matplotlib       : 3.1.2
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
s3fs             : None
scipy            : 1.4.1
sqlalchemy       : 1.3.17
tables           : None
xarray           : None
xlrd             : 1.2.0
xlwt             : None
xlsxwriter       : None

</details>

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #34534

I think the previous description was wrong. We only reindex if the index is not None, not the opposite.

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -180,8 +180,8 @@ class Series(base.IndexOpsMixin, generic.NDFrame):
         Values must be hashable and have the same length as `data`.
         Non-unique index values are allowed. Will default to
         RangeIndex (0, 1, 2, ..., n) if not provided. If data is dict-like
-        and index is None, then the values in the index are used to
-        reindex the Series after it is created using the keys in the data.
+        and index is None, then the keys in the data are used as index. If the
+        index is not None, the resulting Series is reindexed with the index values.
```
Comment (jreback) [inline pandas/core/series.py:184]:
```suggestion
        index is not None, the resulting Series is reindexed with the data values.
```

File: pandas/core/series.py [hunk 2]
```diff
@@ -180,8 +180,8 @@ class Series(base.IndexOpsMixin, generic.NDFrame):
         Values must be hashable and have the same length as `data`.
         Non-unique index values are allowed. Will default to
         RangeIndex (0, 1, 2, ..., n) if not provided. If data is dict-like
-        and index is None, then the values in the index are used to
-        reindex the Series after it is created using the keys in the data.
+        and index is None, then the keys in the data are used as index. If the
```
Comment (jreback) [inline pandas/core/series.py:183]:
```suggestion
        and index is None, then the keys in the data are used as the index. If the
```

File: pandas/core/series.py (hunk 1 above)
Comment (phofl) [reply pandas/core/series.py:184]: I might misunderstand your suggestion, but we create a series which has the dictionary keys as index. After this we perform a reindex operation with index. Basically something like this:

```
data = {"a": 1, "b": 1}
ser = Series(data, index=["x", "y"])
```

This is temporarily equal to
```
a    1
b    1
dtype: int64
```

After this we perform a reindex operation
```
ser.reindex(index)
```
which results in
```
x   NaN
y   NaN
dtype: float64
```

Comment (jreback) [reply pandas/core/series.py:184]: ok made a slightly different workding

File: pandas/core/series.py [hunk 3]
```diff
@@ -180,8 +180,8 @@ class Series(base.IndexOpsMixin, generic.NDFrame):
         Values must be hashable and have the same length as `data`.
         Non-unique index values are allowed. Will default to
         RangeIndex (0, 1, 2, ..., n) if not provided. If data is dict-like
-        and index is None, then the values in the index are used to
-        reindex the Series after it is created using the keys in the data.
+        and index is None, then the keys in the data are used as the index. If the
+        index is not None, the resulting Series is reindexed with the index values.
```
Comment (jreback) [inline pandas/core/series.py:184]:
```suggestion
       passed index keyword to the Series is not None, the resulting Series is reindexed with the passed index values'.
```

Comment (jreback) [reply pandas/core/series.py:184]: actually what you have is fine

**Review by jreback — CHANGES_REQUESTED** [review]
can you add an Example that shows the usage in the OP (and say this in a comment what is happening)

**Comment by phofl** [comment]
Added the example.

I think we should backport here? The sentence I fixed was added in 1.2 and is wrong

File: pandas/core/series.py [hunk 4]
```diff
@@ -190,6 +190,23 @@ class Series(base.IndexOpsMixin, generic.NDFrame):
         The name to give to the Series.
     copy : bool, default False
         Copy input data.
+
+    Examples
+    --------
+    Constructing Series from a dictionary with an Index specified
```
Comment (jreback) [inline pandas/core/series.py:196]: hmm we don't have a good example. can you add the same, but use say index=['a', 'b']) to show what this does *before*

**Comment by jreback** [comment]
> Added the example.
> 
> I think we should backport here? The sentence I fixed was added in 1.2 and is wrong
yup can backport

File: pandas/core/series.py (hunk 4 above)
Comment (phofl) [reply pandas/core/series.py:196]: Yes of course. Used the same values to show the difference

**Review by simonjayhawkins — APPROVED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @phofl

**Comment by jreback** [comment]
@meeseeksdev backport 1.2.x

**Pull request merged**